ddtrace/contrib/internal/vertexai/_utils.py

```diff
@@ -0,0 +1,302 @@
+import sys
+import copy
+
+import vertexai
+from vertexai.generative_models import GenerativeModel, Part
+
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.llmobs._utils import _get_attr
+
+class BaseTracedVertexAIStreamResponse:
+    def __init__(self, generator, instance, integration, span):
+        self._generator = generator
+        self._model_instance = instance
+        self._dd_integration = integration
+        self._dd_span = span
+        self._chunks = []
+
+
+class TracedVertexAIStreamResponse(BaseTracedVertexAIStreamResponse):
+    def __enter__(self):
+        self._generator.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._generator.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        try:
+            for chunk in self._generator.__iter__():
+                self._chunks.append(copy.deepcopy(chunk))
```

Comment:
Why are we deep copying the chunk?

Reply:
I know in the Gemini API, the streamed response actually keeps track and joins the underlying response once the stream is finished (which we just use instead of building the streamed chunks together) is that not the case in VertexAI?

Reply:
yes I noticed that and as far as I could tell Vertex AI does not do the same thing, so I have to keep track of it as I go

Reply:
For `send_message`, the first chunk is actually modified in subsequent iterations which is why I believe the deep copy is necessary in this case.